src/backend/distributed/master/citus_tools.sql

```diff
@@ -0,0 +1,204 @@
+
+CREATE OR REPLACE FUNCTION citus_run_on_all_workers(command text,
+													parallel bool default true,
+													OUT nodename text,
+													OUT nodeport int,
+													OUT success bool,
+													OUT result text)
+	RETURNS SETOF record
+	LANGUAGE plpgsql
+	AS $function$
+DECLARE
+	workers text[];
+	ports int[];
+	commands text[];
+BEGIN
+	WITH citus_workers AS (
+		SELECT * FROM master_get_active_worker_nodes() ORDER BY node_name, node_port)
+	SELECT array_agg(node_name), array_agg(node_port), array_agg(command)
+	INTO workers, ports, commands
+	FROM citus_workers;
+
+	RETURN QUERY SELECT * FROM master_run_on_worker(workers, ports, commands, parallel);
+END;
+$function$;
+
+
+CREATE OR REPLACE FUNCTION citus_run_on_all_placements(table_name regclass, command text,
+													   parallel bool default true,
+													   OUT nodename text,
+													   OUT nodeport int,
+													   OUT shardid bigint,
+													   OUT success bool, OUT result text)
+	RETURNS SETOF record
+	LANGUAGE plpgsql
+	AS $function$
+DECLARE
+	workers text[];
+	ports int[];
+	shards bigint[];
+	commands text[];
+BEGIN
+	WITH citus_placements AS (
+		SELECT
+			ds.logicalrelid::regclass AS tablename,
+			ds.shardid AS shardid,
+			shard_name(ds.logicalrelid, ds.shardid) AS shardname,
+			dsp.nodename AS nodename, dsp.nodeport::int AS nodeport
+		FROM pg_dist_shard ds JOIN pg_dist_shard_placement dsp USING (shardid)
+		WHERE dsp.shardstate = 1 and ds.logicalrelid::regclass = table_name
+		ORDER BY ds.logicalrelid, ds.shardid, dsp.nodename, dsp.nodeport)
+	SELECT
+		array_agg(cp.nodename), array_agg(cp.nodeport), array_agg(cp.shardid),
+		array_agg(format(command, cp.shardname))
+	INTO workers, ports, shards, commands
+	FROM citus_placements cp;
+
+	RETURN QUERY
+		SELECT r.node_name, r.node_port, shards[row_number() over()],
+			r.success, r.result
+		FROM master_run_on_worker(workers, ports, commands, parallel) r;
+END;
+$function$;
+
+
+CREATE OR REPLACE FUNCTION citus_run_on_all_colocated_placements(table_name1 regclass,
+																 table_name2 regclass,
+																 command text,
+																 parallel bool default true,
+																 OUT nodename text,
+																 OUT nodeport int,
+																 OUT shardid1 bigint,
+																 OUT shardid2 bigint,
+																 OUT success bool,
+																 OUT result text)
+	RETURNS SETOF record
+	LANGUAGE plpgsql
+	AS $function$
+DECLARE
+	workers text[];
+	ports int[];
+	shards1 bigint[];
+	shards2 bigint[];
+	commands text[];
+	table_1_shard_count int;
+	table_2_shard_count int;
+BEGIN
+	SELECT COUNT(*) INTO table_1_shard_count FROM pg_dist_shard
+	WHERE logicalrelid::regclass = table_name1;
+
+	SELECT COUNT(*) INTO table_2_shard_count FROM pg_dist_shard
+	WHERE logicalrelid::regclass = table_name2;
+
+	IF (table_1_shard_count != table_2_shard_count) THEN
+		RAISE EXCEPTION 'tables % and % are not co-located', table_name1, table_name2;
+	END IF;
```

Comment:
Hm, this doesn't really guarantee colocation.  Seems better to query something like
```
SELECT count(*) nr_colocated_shards,
    (SELECT count(*) FROM pg_dist_shard a WHERE a.logicalrelid = 'test_table'::regclass) full_shard_count_a,
    (SELECT count(*) FROM pg_dist_shard b WHERE a.logicalrelid = 'test_table2'::regclass) full_shard_count_b
FROM pg_dist_shard tba JOIN pg_dist_shard tbb USING(shardminvalue, shardmaxvalue)
WHERE tba.logicalrelid = 'test_table'::regclass AND tbb.logicalrelid = 'test_table2'::regclass;
```